### PR TITLE
Replace health factor with collateral ratio, test fixes, more natspec

### PR DESCRIPTION
**Pull request by AmadiMichael**
- Replace use of health factor with collateral ratio
   - This does not change any thing for integrators, it's just a more optimised and straightforward way to perform internal (within vault) solvency check. The periphery will also return true or false on it's end so it's also just an internal implementation detail on that part too
- fix wrong emitted event data and add tests for it,
- add more natspec
- use GPL-3.0 license

### CODE REVIEW

File: src/interfaces/IVault.sol [hunk 1]
```diff
@@ -5,14 +5,14 @@ interface IVault {
     // ------------------------------------------------ CUSTOM ERROR ------------------------------------------------
     error ZeroAddress();
     error UnrecognizedParam();
-    error BadHealthFactor();
```
Comment (njokuScript) [inline src/interfaces/IVault.sol:8]: can we still keep the name HealthFactor? Much clearer

File: src/interfaces/IVault.sol [hunk 2]
```diff
@@ -5,14 +5,14 @@ interface IVault {
...
+    error BadCollateralRatio();
     error PositionIsSafe();
     error ZeroCollateral();
     error TotalUserCollateralBelowFloor();
     error CollateralAlreadyExists();
     error CollateralDoesNotExist();
     error NotOwnerOrReliedUpon();
-    error HealthFactorNotImproved();
```
Comment (njokuScript) [inline src/interfaces/IVault.sol:15]: same here

**Review by njokuScript — COMMENTED** [review]
Can you also specify how this changes affects our documentation on Health factor

File: src/interfaces/IVault.sol (hunk 2 above)
Comment (njokuScript) [reply src/interfaces/IVault.sol:15]: can you also add a GPL-3.0 license to the project?

File: src/interfaces/IVault.sol (hunk 1 above)
Comment (AmadiMichael) [reply src/interfaces/IVault.sol:8]: I actually think collateral ratio is much clearer as it's easy to understand that if it's higher than the liquidation threshold then it's a bad value. Health factor would need a new reader to look for what that means first, it's a whole new phrase not used anywhere else in the codebase too.

File: src/interfaces/IVault.sol (hunk 2 above)
Comment (AmadiMichael) [reply src/interfaces/IVault.sol:15]: > can you also add a GPL-3.0 license to the project?

done

**Comment by AmadiMichael** [comment]
> Can you also specify how this changes affects our documentation on Health factor

done

**Comment by njokuScript** [comment]
> > Can you also specify how this changes affects our documentation on Health factor
> 
> done

where did you add this?

**Comment by AmadiMichael** [comment]
> > > Can you also specify how this changes affects our documentation on Health factor
> > 
> > done
> 
> where did you add this?

In the pr description

**Comment by AmadiMichael** [comment]
> > > Can you also specify how this changes affects our documentation on Health factor
> > 
> > 
> > done
> 
> where did you add this?

in the pr description

**Pull request merged**